opentracing/src/main/java/opentracing/Span.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+/**
+ * Represents an in-flight span in the opentracing system.
+ *
+ * <p>Spans are created by the {@link Tracer#buildSpan} interface.
+ */
+public interface Span {
+
+  /**
+   * Sets the end timestamp and records the span.
+   *
+   * <p>This should be the last call made to any span instance, and to do otherwise leads to
+   * undefined behavior.
+   */
+  void finish();
+
+  /**
+   * Set a key:value tag on the Span.
+   *
+   * <p>If there is a pre-existing tag set for {@code key}, it is overwritten.
+   */
+  // overloaded 3x to support the BasicType concern
+  Span setTag(String key, String value);
+
+  /** Same as {@link #setTag(String, String)}, but for boolean values. */
+  Span setTag(String key, boolean value);
+
+  /** Same as {@link #setTag(String, String)}, but for numeric values. */
+  Span setTag(String key, Number value);
+
+  /**
+   * Set a Baggage item, represented as a simple string:string pair.
```

Comment:
I see you have left out the unpalatable HTTP-header-inspired restrictions on the `key` here... intentional? It is a deviation from the spec in any case.